backend_new/test/integration/ami-chart.e2e-spec.ts

```diff
@@ -0,0 +1,189 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from '../../src/app.module';
+import { PrismaService } from '../../src/services/prisma.service';
+import { jurisdictionFactory } from '../../prisma/seed-helpers/jurisdiction-factory';
+import { stringify } from 'qs';
+import { AmiChartQueryParams } from '../../src/dtos/ami-charts/ami-chart-query-params.dto';
+import { amiChartFactory } from '../../prisma/seed-helpers/ami-chart-factory';
+import { AmiChartCreate } from '../../src/dtos/ami-charts/ami-chart-create.dto';
+import { AmiChartUpdate } from '../../src/dtos/ami-charts/ami-chart-update.dto';
+import { IdDTO } from 'src/dtos/shared/id.dto';
+
+describe('AmiChart Controller Tests', () => {
+  let app: INestApplication;
+  let prisma: PrismaService;
+
+  beforeEach(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+    await app.init();
+  });
+
+  const cleanUpDb = async (
+    amiChartIds: string[],
+    jurisdictionIds: string[],
+  ) => {
+    for (let i = 0; i < amiChartIds.length; i++) {
+      await prisma.amiChart.delete({
+        where: {
+          id: amiChartIds[i],
+        },
+      });
+    }
+    for (let i = 0; i < jurisdictionIds.length; i++) {
+      await prisma.jurisdictions.delete({
+        where: {
+          id: jurisdictionIds[i],
+        },
+      });
+    }
+  };
+
+  it('testing list endpoint', async () => {
+    const jurisdictionA = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(10),
+    });
+    const jurisdictionB = await prisma.jurisdictions.create({
+      data: jurisdictionFactory(11),
+    });
+    const amiChartA = await prisma.amiChart.create({
+      data: amiChartFactory(10, jurisdictionA.id),
+    });
+    const amiChartB = await prisma.amiChart.create({
+      data: amiChartFactory(15, jurisdictionB.id),
+    });
+    const queryParams: AmiChartQueryParams = {
+      jurisdictionId: jurisdictionA.id,
+    };
+    const query = stringify(queryParams as any);
+
+    const res = await request(app.getHttpServer())
+      .get(`/amiCharts?${query}`)
+      .expect(200);
+
+    expect(res.body.length).toEqual(1);
+    expect(res.body[0].name).toEqual(amiChartA.name);
+
+    await cleanUpDb(
```

Comment:
recommend that this call gets moved to an afterEach, so a dev doesn't have to remember to add it to each test

Reply:
moved!

Reply:
I don't see this updated?

Reply:
sorry yeah I tried to leave a note in a couple places, moving this into afterEach was running into race cases so I set it back